### ISSUE TITLE
size of fetched data reduces by 1 per request

### ISSUE THREAD

**Issue by Spanoudakhs** [pushshift/api#32]
Every next request I get one less result .
Use [this](https://raw.githubusercontent.com/Spanoudakhs/JSSR/master/bug.js) in your console to test it .

**Comment by iamgroot42** [comment]
@Spanoudakhs could you please give a new link? The file (the whole repository, in fact) seems to be 404

**Comment by ghost** [comment]
@iamgroot42 I have tried to reproduce from what I describe on my issue and I can not reproduce it . Here is an example that works with no issues (when the server of pushshift has no issues): 
```js
(async () => {
	let ti = Date.UTC(2018, 6 - 1, 1) / 1000;
	const tf = Date.UTC(2019, 1 - 1, 1) / 1000;
	const allData = [];
	let newData;
	let url;
	const size = 100;
	do {
		url = `https://api.pushshift.io/reddit/search/comment/?q=jQuery&subreddit=webdev&after=${ti}&before=${tf}&size=${size}&fields=created_utc`;
		newData = (await fetch(url).then((data) => data.json())).data;
		allData.push(...newData);
		ti = allData[allData.length - 1].created_utc;
		console.log(url, allData.length);
	} while (newData.length === size);
})();
```
I was noob back then . I did not know how to write properly js in markdown and that is why I created a repository with js file that had the code that reproduced the bug.

Do you happen to have any similar bug ? 

I think the issue should not be closed but deleted since there is no code that reproduces it and me , the author of the issue , can not reproduce it anymore .

**Comment by iamgroot42** [comment]
Yeah, I did. It was because the documentation page says that comments are sorted by time with the latest ones **first**, but the API actually returns the oldest ones **first**. I assumed proper behavior and simply used the first post's created_utc for pagination, whereas it should have been the last post. As a result, I was effectively making the same query again and again, with >=1 result less (since it excludes posts on that exact timestamp).
